src/Arriba/Arriba.Server/Program.cs

```diff
@@ -1,40 +1,43 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
 using Arriba.Configuration;
+using Arriba.Diagnostics.Tracing;
 using Arriba.Monitoring;
 using Microsoft.AspNetCore.Builder;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.Extensions.Hosting;
-using System;
-using System.Diagnostics;
 using AspNetHost = Microsoft.Extensions.Hosting.Host;
 
 namespace Arriba.Server
 {
-    internal class Program
+    internal class ArribaServer
     {
-        private static void Main(string[] args)
+        private static async Task Main(string[] args)
         {
-            Console.WriteLine("Arriba Local Server\r\n");
-
-            var configLoader = new ArribaConfigurationLoader(args);
-
-            // Write trace messages to console if /trace is specified 
-            if (configLoader.GetBoolValue("trace", Debugger.IsAttached))
+            await ArribaProgram.Run<ArribaServer>(() =>
             {
-                EventPublisher.AddConsumer(new ConsoleEventConsumer());
-            }
+                var configLoader = new ArribaConfigurationLoader(args);
+
+                // Write trace messages to console if /trace is specified 
+                if (configLoader.GetBoolValue("trace", Debugger.IsAttached))
+                {
+                    EventPublisher.AddConsumer(new ConsoleEventConsumer());
+                }
 
-            // Always log to CSV
-            EventPublisher.AddConsumer(new CsvEventConsumer());
+                // Always log to CSV
+                EventPublisher.AddConsumer(new CsvEventConsumer());
 
-            Trace.Listeners.Add(new TextWriterTraceListener(Console.Out));
+                Trace.Listeners.Add(new TextWriterTraceListener(Console.Out));
 
-            CreateHostBuilder(args).Build().Run();
+                CreateHostBuilder(args).Build().Run();
 
-            Console.WriteLine("Exiting.");
-            Environment.Exit(0);
+                Environment.Exit(0);
+                return Task.CompletedTask;
```

Comment:
Is this `return.CompletedTask;` only necessary because the method was changed to async Task and we need it to compile? I thought this line would never execute due to `Environment.Exit(0);` before it.

Reply:
Why are we even calling Environment.Exit(0)?
All threads should gracefully terminate once they are done - this will hide any runaway thread if any...